From d0b790dcd02da959cbdfc83d606906cead9e8375 Mon Sep 17 00:00:00 2001
From: David Faure <david.faure@kdab.com>
Date: Sat, 12 Apr 2014 11:25:28 +0200
Subject: [PATCH 10/37] QDbus: Fix (!a == b) comparison

! binds to a, and that is wrong here.

(cherry picked from qtbase/4b7cd57719a637189696d673b014ae785df669bf)

Change-Id: I75542a0c27f39fb6e684dedd9925a1f3748d4919
Reviewed-by: Thiago Macieira <thiago.macieira@intel.com>
---
 src/dbus/qdbuspendingcall.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/dbus/qdbuspendingcall.cpp b/src/dbus/qdbuspendingcall.cpp
index 06597a6..b7def2e 100644
--- a/src/dbus/qdbuspendingcall.cpp
+++ b/src/dbus/qdbuspendingcall.cpp
@@ -225,7 +225,7 @@ void QDBusPendingCallPrivate::checkReceivedSignature()
         return;                 // no signature to validate against
 
     // can't use startsWith here because a null string doesn't start or end with an empty string
-    if (!replyMessage.signature().indexOf(expectedReplySignature) == 0) {
+    if (replyMessage.signature().indexOf(expectedReplySignature) != 0) {
         QString errorMsg = QLatin1String("Unexpected reply signature: got \"%1\", "
                                          "expected \"%2\"");
         replyMessage = QDBusMessage::createError(
-- 
1.9.3

